apollo-router/src/plugins/connectors/make_requests.rs

```diff
@@ -37,33 +38,47 @@ pub(crate) enum ResponseKey {
     RootField {
         name: String,
         typename: ResponseTypeName,
-        #[allow(dead_code)]
         selection_set: apollo_compiler::executable::SelectionSet,
+        inputs: Value,
     },
     Entity {
         index: usize,
         typename: ResponseTypeName,
-        #[allow(dead_code)]
         selection_set: apollo_compiler::executable::SelectionSet,
+        inputs: Value,
     },
     EntityField {
         index: usize,
         field_name: String,
         typename: ResponseTypeName,
-        #[allow(dead_code)]
         selection_set: apollo_compiler::executable::SelectionSet,
+        inputs: Value,
```

Comment:
I'll need to look into this a bit. It was expanding on a pattern Lenny had used, but I need to look into the implications.

Reply:
Ok, I thought this was the case but wanted to confirm. This `ResponseKey` was originally just a key (typename and field name) indicating the location where the response from the REST call should be inserted into the GraphQL response. It's kind of grown to really be more of a `ResponseContext` (in fact it might be worth a rename), where any data needed to process the REST response is stored. It's really just a way to transfer data between the code creating the request and the code processing the response. It's inserted as an "extension" but it really isn't processed as an HTTP extension - it's just using that extension field as a way to store arbitrary data so it can be picked up by the code handling the response. It's stored *after* the REST call returns, and not sent in the REST request. It's then read back out in `handle_responses.rs` when processing the REST response. The data is used locally there, and discarded.